addons/mail/static/tests/suggestion/suggestion.test.js

```diff
@@ -91,6 +91,25 @@ test('display partner mention suggestions on typing "@" in chatter', async () =>
     await contains(".o-mail-Composer-suggestion strong", { text: "Mitchell Admin" });
 });
 
+test("Do not fetch if search more specific and fetch had no result", async () => {
+    await startServer();
+    onRpc("res.partner", "get_mention_suggestions", () => {
+        step("get_mention_suggestions");
+    });
```

Comment:
Ideally should add control with promise and also ensure no extra step while still typing more specific during the RPC.

This won't work with current version of the code, because `lastEmptySearch` should be based on the result of the RPC, not on the following update/state.items that are based on current term (and not term used for the RPC, which is inconsistent).